File: BackEnd/eMenka/eMenka.Domain/Classes/Corporation.cs
using System;

namespace eMenka.Domain.Classes
{
    public class Corporation
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Abbreviation { get; set; }
        public Company Company { get; set; }
        public int CompanyId { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime? EndDate { get; set; }
    }
}

File: eFenkaFrontend/src/app/models/corporation/corporation.ts
import { Company } from '../company/company';

export class Corporation {
  id: number;
  name: string;
  abbreviation: string;
  company: Company;
  startDate: Date;
  endDate: Date;

  constructor(id: number, name: string, abbreviation: string, company: Company, startDate: Date, endDate: Date) {
    this.id = id;
    this.name = name;
    this.abbreviation = abbreviation;
    this.company = company;
    this.startDate = startDate;
    this.endDate = endDate;
  }
}
